File: code/search/linear_search/LinearSearch.cs
// Part of Cosmos by OpenGenus
public class LinearSearch
{
  public static void Main(string[] args)
  {
    int[] items = new int[] { 10, 12, 52, 634, 24, 743, 234, 7, 25, 742, 76, 25};
    
    int itemToFind = 634;
    int itemIndex = -1;
    
    for(int i = 0; i < items.Length; i++)
    {
      if(items[i] == itemToFind)
      {
          itemIndex = i;
          break;
      }
    }
    
    if(itemIndex != -1)
     System.Console.WriteLine(itemToFind + " found at " + itemIndex);
    else
    System.Console.WriteLine(itemToFind + " not found!");
  }
}

File: code/search/src/linear_search/linear_search.rb
# Part of Cosmos by OpenGenus Foundation

items = [5, 10, 34, 18, 6, 7, 45, 67]
target = 7

##
# Returns a boolean that indicates whether the target item is contained within
# the given list.
def linear_search(list, target)
  # Iterate over each item in the list
  list.each do |item|
    return true if item == target
  end

  false
end

found = linear_search(items, target)

if found
  puts "Found '#{target}'!"
else
  puts "'#{target}' was not found."
end
